android/src/main/java/com/marigold/rnsdk/RNEngageBySailthruModule.kt

```diff
@@ -225,8 +232,10 @@ class RNEngageBySailthruModule (reactContext: ReactApplicationContext) : ReactCo
     }
 
     //Helper method for instantiating EngageBySailthru
-    fun createEngageBySailthru(): EngageBySailthru {
-        return EngageBySailthru()
+    fun createEngageBySailthru(): EngageBySailthru? = try {
```

Comment:
In order to provide some feedback to the developer, we could adjust this slightly:
```
fun createEngageBySailthru(promise: Promise? = null): EngageBySailthru? = try {
    EngageBySailthru()
} catch (e: Exception) {
    promise?.reject(ERROR_CODE_KEY,  e.message)
    null
}
```
and pass the promise in for methods that handle them.